vcert/connection_tpp_token.py

```diff
@@ -18,77 +18,107 @@
 
-PATH_AUTHORIZE_TOKEN = API_TOKEN_URL + "authorize/oauth"  # type: str
-PATH_REFRESH_TOKEN = API_TOKEN_URL + "authorize/token"  # type: str
-PATH_REVOKE_TOKEN = API_TOKEN_URL + "revoke/token"  # type: str
-PATH_CERTIFICATE_GUID = API_BASE_URL + "certificate"  # type: str
+from .common import MIME_JSON, TokenInfo, Authentication, CommonConnection, KeyType, Policy, ZoneConfig, CertField
+from .errors import (ClientBadData, ServerUnexptedBehavior, AuthenticationError, CertificateRequestError,
+                     CertificateRenewError)
+from .pem import parse_pem
 
 HEADER_AUTHORIZATION = "Authorization"  # type: str
 
-KEY_ACCESS_TOKEN = "access_token"  # type: str
-KEY_REFRESH_TOKEN = "refresh_token"  # type: str
+KEY_ACCESS_TOKEN = "access_token"  # type: str  # nosec
+KEY_REFRESH_TOKEN = "refresh_token"  # type: str  # nosec
 KEY_EXPIRATION_DATE = "expiration_date"  # type: str
 
 
-class TPPTokenConnection(TPPCommonConnection):
+class URLS:
+    API_TOKEN_URL = "vedauth/"  # type: str  # nosec
+    API_BASE_URL = "vedsdk/"  # type: str  # nosec
+
+    AUTHORIZE_TOKEN = API_TOKEN_URL + "authorize/oauth"  # type: str
+    REFRESH_TOKEN = API_TOKEN_URL + "authorize/token"  # type: str
+    REVOKE_TOKEN = API_TOKEN_URL + "revoke/token"  # type: str
+
+    AUTHORIZE = API_BASE_URL + "authorize/"
+    CERTIFICATE_REQUESTS = API_BASE_URL + "certificates/request"
+    CERTIFICATE_RETRIEVE = API_BASE_URL + "certificates/retrieve"
+    FIND_POLICY = API_BASE_URL + "config/findpolicy"
+    CERTIFICATE_REVOKE = API_BASE_URL + "certificates/revoke"
+    CERTIFICATE_RENEW = API_BASE_URL + "certificates/renew"
+    CERTIFICATE_SEARCH = API_BASE_URL + "certificates/"
+    CERTIFICATE_IMPORT = API_BASE_URL + "certificates/import"
+    ZONE_CONFIG = API_BASE_URL + "certificates/checkpolicy"
+    CONFIG_READ_DN = API_BASE_URL + "Config/ReadDn"
+
+
+class TPPTokenConnection(CommonConnection):
     def __init__(self, url, user=None, password=None, access_token=None, refresh_token=None, http_request_kwargs=None):
-        super().__init__(http_request_kwargs=http_request_kwargs)
+        """
+        :param str url:
+        :param str user:
+        :param str password:
+        :param str access_token:
+        :param str refresh_token:
+        :param dict[str,Any] http_request_kwargs:
+        """
         self._base_url = url  # type: str
         self._auth = Authentication(user=user, password=password, access_token=access_token,
                                     refresh_token=refresh_token)  # type: Authentication
+        if http_request_kwargs is None:
+            http_request_kwargs = {"timeout": 180}
+        elif "timeout" not in http_request_kwargs:
+            http_request_kwargs["timeout"] = 180
+        self._http_request_kwargs = http_request_kwargs or {}
+
+    def __setattr__(self, key, value):
+        if key == "_base_url":
+            value = self._normalize_and_verify_base_url(value)
+        self.__dict__[key] = value
```

Comment:
There is a Python wrapper called @property. Its purpose is to define a getter, setter, and/or deleter method to an instance or class variable. It's nicer for refactoring, mostly. Worth noting if you're interested.